packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -518,6 +518,11 @@ function unmount(
             break;
         }
 
+        case VNodeType.Fragment: {
+            unmountVNodes(vnode.children, elm as ParentNode, renderer, false);
+            break;
+        }
```

Comment:
In the `unmount` invocations, after the `doRemove` iteration, the vnodes are manually removed. 

Since there isn't a case for `VNodeType.Fragment`, it does not disconnect them.

This isn't an issue when `ENABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE` is true because they will be invoked on `renderer.remove`.